src/abi/typeFormulaParser.ts

```diff
@@ -81,9 +81,10 @@ export class TypeFormulaParser {
     private acquireTypeWithParameters(stack: any[]): TypeFormula {
         const typeParameters = this.acquireTypeParameters(stack);
         const typeName = stack.pop();
+        const metadata = typeParameters[0].name;
```

Comment:
To be more explicit that we deal with metadata only for some types, variable should be defined under the scope of `if managed decimal` (I know it works the same).

Optional, can also stay as it is.